app/controllers/unattended_controller.rb

```diff
@@ -92,6 +138,7 @@ def render_template(type)
   def get_host_details
     @host = find_host_by_spoof || find_host_by_token
     @host ||= find_host_by_ip_or_mac unless token_from_params.present?
+    return if ipxe_request?
     verify_valid_host_token
     verify_found_host
   end
```

Comment:
I have to be missing something, you skip these checks and perform it in the newly created method. It reads awkward, maybe there is something I am missing here.

Reply:
Those verifications need to be skipped in case the request comes from a non-existant host.

When I wrote this code, I didn't want to have to do my own host discovery, but as the template rendering code also needs to trigger even if a host is *not* discovered - for global template rendering - I had to skip the verifications at this point.

But yes, it reads a bit awkwardly.

Reply:
We should be running those validations in a single place rather than branching like that. You can easily tell when not to perform those (it is an iPXE request without mac, token or http mac parameter).

Reply:
@leewaa and I started refactoring the whole controller. Stay tuned. It's a lot of work but you it's clearly worth it.